pkg/test/framework/components/echo/kube/deployment.go

```diff
@@ -184,11 +187,102 @@ data:
 ---
 {{- end}}
 `
+
+	// vmDeploymentYaml aims to simulate a VM, but instead of managing the complex test setup of spinning up a VM,
+	// connecting, etc we run it inside a pod. The pod has pretty much all Kubernetes features disabled (DNS and SA token mount)
+	// such that we can adequately simulate a VM and DIY the bootstrapping.
+	vmDeploymentYaml = `
+{{- $cluster := .Cluster }}
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ $.Service }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      istio.io/test-vm: {{ $.Service }}
+  template:
+    metadata:
+      annotations:
+        # Sidecar is inside the pod to simulate VMs - do not inject
+        sidecar.istio.io/inject: "false"
+      labels:
+        # Label should not be selected. We will create a workload entry instead
+        istio.io/test-vm: {{ $.Service }}
+    spec:
+      # Disable kube-dns, to mirror VM
+      dnsPolicy: Default
+      # Disable service account mount, to mirror VM
+      automountServiceAccountToken: false
+      containers:
+      - name: istio-proxy
+        image: {{ $.Hub }}/app_sidecar:{{ $.Tag }}
+        #image: {{ $.Hub }}/app_sidecar:{{ $.Tag }}
+        imagePullPolicy: {{ $.PullPolicy }}
+        securityContext:
+          capabilities:
+            add:
+            - NET_ADMIN
+          runAsUser: 1338
+          runAsGroup: 1338
+        command:
+        - bash
+        - -c
+        - |-
+          sudo sh -c 'echo ISTIO_SERVICE_CIDR=* > /var/lib/istio/envoy/cluster.env'
+          sudo sh -c 'echo ISTIO_PILOT_PORT={{$.VM.IstiodPort}} >> /var/lib/istio/envoy/cluster.env'
+          sudo sh -c 'echo "{{$.VM.IstiodIP}} istiod.istio-system.svc" >> /etc/hosts'
+          sudo sh -c 'echo "1.1.1.1 pod.{{$.Namespace}}.svc.cluster.local" >> /etc/hosts'
```

Comment:
So this is assuming that everyone tinkers with etc/hosts to make k8s internal services reachable from VMs - which is definitely not the case. The more common case is one where people access the hostname exposed on Gateway. But since this traffic is internal, by creating a virtual service bound to the `mesh` gateway, we essentially forward to the clusters that correspond to the K8S services [but still no DNS resolution].

Reply:
There are many ways to run VMs. Right now we have 0 testing, this adds a test for one model. As we improve DNS resolution we will throw out the hacky etc/hosts stuff. And we can also add the gateway mode as well. People accessing through the gateway doesn't mean we should *not* test this, it means we should test both

Reply:
@rshriram agree modifying /etc/hosts is not good. The proper approach was covered in a prior review (somewhere) by @smawson. I believe his proposal was that we add istio's coredns in the search list in /etc/resolv.conf. Then anything port 53 would be captured via the sidecar and forwarded to coredns.

Your proposal also makes sense. This coming week I am going to take a look at implementing proper DNS resolution for VMs.

In general though, having some tests is better than having none. If we can agree on a canonical way to have proper DNS resolution within a VM, then we can focus testing energy on the one method (I don't really see people modifying /etc/hosts, this is just what is currently documented).

Cheers,
-steve